pdns/backends/gsql/gsqlbackend.cc

```diff
@@ -1265,12 +1292,24 @@ void GSQLBackend::getAllDomains(vector<DomainInfo> *domains, bool include_disabl
       } catch (...) {
         continue;
       }
+
+      if (pdns_iequals(row[3], "MASTER"))
+        di.kind = DomainInfo::Master;
+      else if (pdns_iequals(row[3], "SLAVE"))
+        di.kind = DomainInfo::Slave;
+      else
+        di.kind = DomainInfo::Native;
   
       if (!row[4].empty()) {
         vector<string> masters;
         stringtok(masters, row[4], " ,\t");
-        for(const auto& m : masters)
-          di.masters.emplace_back(m, 53);
+        for(const auto& m : masters) {
+          try {
+            di.masters.emplace_back(m, 53);
+          } catch(const PDNSException &e) {
+            throw PDNSException("Could not parse master address (" + m + ") for zone '" + di.zone.toLogString() + "': " + e.reason);
```

Comment:
I'm prety sure this throw will lead to new issues since a single broken master ip will terminate getAllDomans(). 
For example "pdnsutil rectify-all-zones" in a cron, will stop working if there is one broken master ip in your database.

Reply:
`rectify-all-zones` works, but `list-all-zones` and `check-all-zones` break indeed. Not throwing will make `pdnsutil` swallow the exception (but `check-all-zones` will report it). Fixed